pkg/svcat/service-catalog/plan.go

```diff
@@ -34,21 +34,36 @@ const (
 
 // RetrievePlans lists all plans defined in the cluster.
 func (sdk *SDK) RetrievePlans(opts *FilterOptions) ([]v1beta1.ClusterServicePlan, error) {
-	plans, err := sdk.ServiceCatalog().ClusterServicePlans().List(v1.ListOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("unable to list plans (%s)", err)
+	// Filter by Broker
+	if opts.Broker != "" {
+		brokerOpts := v1.ListOptions{
+			FieldSelector: fields.OneTermEqualSelector(FieldClusterServiceBrokerName, opts.Broker).String(),
+		}
+		plansByBroker, err := sdk.ServiceCatalog().ClusterServicePlans().List(brokerOpts)
+		if err != nil {
+			return nil, fmt.Errorf("unable to list plans by broker (%s)", err)
+		} else {
+			return plansByBroker.Items, nil
+		}
 	}
 
-	if opts != nil && opts.ClassID != "" {
-		plansFiltered := make([]v1beta1.ClusterServicePlan, 0)
-		for _, p := range plans.Items {
-			if p.Spec.ClusterServiceClassRef.Name == opts.ClassID {
-				plansFiltered = append(plansFiltered, p)
-			}
+	// Filter by ClassID
```

Comment:
Same feedback as above. Please collapse this down so that the call to list isn't done in the if/else.